consensus/common/src/main/java/org/hyperledger/besu/consensus/common/MigratingMiningCoordinator.java

```diff
@@ -118,13 +125,13 @@ public void onBlockAdded(final BlockAddedEvent event) {
         miningCoordinatorSchedule.getFork(currentBlock + 1).getValue();
     if (activeMiningCoordinator != nextMiningCoordinator) {
       LOG.trace(
-          "Switching mining coordinator at block {} from {} to {}",
+          "Migrating mining coordinator at block {} from {} to {}",
           currentBlock,
           activeMiningCoordinator.getClass().getSimpleName(),
           nextMiningCoordinator.getClass().getSimpleName());
       activeMiningCoordinator.stop();
-      nextMiningCoordinator.start();
       activeMiningCoordinator = nextMiningCoordinator;
+      startActiveMiningCoordinator();
     }
     if (activeMiningCoordinator instanceof BlockAddedObserver) {
       ((BlockAddedObserver) activeMiningCoordinator).onBlockAdded(event);
```

Comment:
From my local testing I don't think we need the trigger the onBlockAdded manually. The delegated miningCoordinators are able to subscribe to the onBlockAdded and appear to be working without this.

So if we remove propagating this event then the need to remove the observer will also not be needed.

Reply:
After some discussion on this, this approach of not removing the observer from the delegate mining coordinator and relying on it's on block added subscription has some pitfalls. The one being we don't know which observer will be called first and if there will be a delay between the two. It could be possible the previous mining co-ordinator is able to send a proposal even.

So don't think this is a good idea afterall.